src/compiler/utilities.rs

```diff
@@ -36,39 +86,100 @@ pub fn get_binary_operator_precedence(kind: SyntaxKind) -> OperatorPrecedence {
     OperatorPrecedence::Invalid
 }
 
+pub fn create_diagnostic_collection() -> DiagnosticCollection {
+    DiagnosticCollection::new()
+}
+
+impl DiagnosticCollection {
+    pub fn new() -> Self {
+        DiagnosticCollection {
+            file_diagnostics: HashMap::<String, SortedArray<Rc<Diagnostic>>>::new(),
+        }
+    }
+
+    pub fn add(&mut self, diagnostic: Rc<Diagnostic>) {
+        // let diagnostics = self
```

Comment:
This is another place where the borrow checker was unhappy with this way of writing the logic